### PR TITLE
Use tokio-compat for 0.2 timers in stdio crate

### PR DESCRIPTION
**Pull request by tomhoule**
Use-case: we rely on tokio 0.2 timers in our application code, but the stdio server crate runs the futures on the 0.1 executor. With tokio-compat, we should be able to get the best of both words, until `jsonrpc-core` works with `std::future::Future`.

### CODE REVIEW

**Comment by tomhoule** [comment]
I can't access the results of the failing CI check, so I don't know what's failing at the moment :)

File: stdio/Cargo.toml [hunk 1]
```diff
@@ -11,12 +11,12 @@ version = "14.0.1"
 
 [dependencies]
 futures = "0.1.23"
-jsonrpc-core = { version = "14.0", path = "../core" }
```
Comment (tomusdrw) [inline stdio/Cargo.toml]: Please revert this change, we want the local crates to use local dependencies, not go to crates.io every time.

**Comment by tomhoule** [comment]
Both changes made :)

**Comment by tomhoule** [comment]
Update on this: we came up with an alternate solution with our own stdio adapter, feel free to merge or close this. For us to return to the jsonrpc-stdio-server crate it would need return a future instead of blocking, and be on tokio 0.2.

**Comment by tomusdrw** [comment]
Transition to `tokio=0.2` and latest futures is a bigger task that will most likely change more things than this. I think it all should be addressed together.
